luarules/gadgets/game_quick_start.lua

```diff
@@ -0,0 +1,804 @@
+local optionsToNodeType = config.optionsToNodeType
+local unitDefs = UnitDefs
+local unitDefNames = UnitDefNames
+
+local gameFrameTryCount = 0
+local initialized = false
+local isGoodWind = false
+local isMetalMap = false
+local metalSpotsList = nil
+local running = false
+
+local allTeamsList = {}
+local boostableCommanders = {}
+local commanders = {}
+local defMetergies = {}
+local commanderFactoryDiscounts = {}
+local mexDefs = {}
+local optionDefIDToTypes = {}
+local queuedCommanders = {}
+
+local function getQuotas(isMetalMap, isInWater, isGoodWind)
+	return config.quotas[isMetalMap and "metalMap" or "nonMetalMap"][isInWater and "water" or "land"]
+	[isGoodWind and "goodWind" or "badWind"]
+end
+
+local function generateLocalGrid(commanderID)
+	local comData = commanders[commanderID]
+	local originX, originZ = comData.spawnX, comData.spawnZ
+	local buildDefID
+	if comData.isInWater then
+		buildDefID = comData.buildDefs and comData.buildDefs.tidal or nil
+	else
+		buildDefID = comData.buildDefs and comData.buildDefs.windmill or nil
+	end
+	if not buildDefID then
+		return {}
+	end
+	local dx = MAP_CENTER_X - originX
+	local dz = MAP_CENTER_Z - originZ
+	local skipDirection = math.abs(dx) >= math.abs(dz) and "x" or "z"
+	local maxOffset = BASE_GENERATION_RANGE
+	local gridList = {}
+	local used = {}
+	local noGoZones = {}
+	table.insert(noGoZones, { x = originX, z = originZ, distance = 100 })
+	if comData.nearbyMexes then
+		for i = 1, #comData.nearbyMexes do
+			local mex = comData.nearbyMexes[i]
+			table.insert(noGoZones, { x = mex.x, z = mex.z, distance = BUILD_SPACING })
+		end
+	end
+	for offsetX = -maxOffset, maxOffset, BUILD_SPACING do
+		for offsetZ = -maxOffset, maxOffset, BUILD_SPACING do
+			local index = (skipDirection == "x" and offsetZ or offsetX) + maxOffset
+			if (index / BUILD_SPACING) % SKIP_STEP ~= 0 then
+				local testX = originX + offsetX
+				local testZ = originZ + offsetZ
+				if math.distance2d(testX, testZ, originX, originZ) <= BASE_GENERATION_RANGE then
+					local tooClose = false
+					for i = 1, #noGoZones do
+						local g = noGoZones[i]
+						if math.distance2d(testX, testZ, g.x, g.z) <= g.distance then
+							tooClose = true
+							break
+						end
+					end
+					if not tooClose then
+						local searchY = spGetGroundHeight(testX, testZ)
+						local snappedX, snappedY, snappedZ = spPos2BuildPos(buildDefID, testX, searchY, testZ)
+						if snappedX and spTestBuildOrder(buildDefID, snappedX, snappedY, snappedZ, 0) == BUILD_ORDER_FREE then
+							local key = snappedX .. "_" .. snappedZ
+							if not used[key] then
+								used[key] = true
+								table.insert(gridList, { x = snappedX, z = snappedZ })
+							end
+						end
+					end
+				end
+			end
+		end
+	end
+	return gridList
+end
+
+for unitDefID, unitDef in pairs(unitDefs) do
+	local metalCost, energyCost = unitDef.metalCost or 0, unitDef.energyCost or 0
+	defMetergies[unitDefID] = metalCost + (energyCost / ENERGY_VALUE_CONVERSION_DIVISOR)
```

Comment:
IDK. I feel like the creative factor here might be hard to explain/too opinionated. It's giving inherent power to wind builds for example. If my energy is suddenly 1/10th cost in the pre-game that has some implications for builds/my blueprint choices. I think it might be better to just agree on a reasonable bound for metal, and a reasonable bound of energy, and treat them as separate thresholds.